rootex/framework/components/visual/model/model_component.cpp

```diff
@@ -109,11 +109,11 @@ void ModelComponent::assignOverrides(Ref<ModelResourceFile> newModel, const Hash
 	m_MaterialOverrides.clear();
 	for (auto& [material, meshes] : m_ModelResourceFile->getMeshes())
 	{
-		setMaterialOverride(material, material);
+		setMaterialOverride(material.get(), material);
```

Comment:
Changing ownership policies like this should be thoroughly tested and scrutinised. This can give birth to tons of bugs if not done right.

Reply:
has been tested through various scenes :heavy_check_mark: